Source/JavaScriptCore/dfg/DFGInPlaceAbstractState.cpp

```diff
@@ -421,14 +418,289 @@ inline bool InPlaceAbstractState::mergeToSuccessors(BasicBlock* basicBlock)
         ASSERT(basicBlock->cfaBranchDirection == InvalidBranchDirection);
         return merge(basicBlock, terminal->targetBlock());
     }
-        
+
     case Branch: {
         ASSERT(basicBlock->cfaBranchDirection != InvalidBranchDirection);
         bool changed = false;
-        if (basicBlock->cfaBranchDirection != TakeFalse)
+        using ValueVector = Vector<std::tuple<Node*, AbstractValue>, 2>;
+        using VariableVector = Vector<std::tuple<Operand, AbstractValue>, 2>;
+        ValueVector trueNodes;
+        ValueVector falseNodes;
+
+        auto collectSparseConditionalPropagationValues = recursableLambda([&](auto self, Edge edge, ValueVector& trueNodes, ValueVector& falseNodes) -> void {
+            auto constant = [&](Node* node, JSValue constant) {
+                AbstractValue value;
+                value.set(m_graph, *m_graph.freeze(constant), basicBlock->cfaStructureClobberStateAtTail);
+                value.fixTypeForRepresentation(m_graph, node);
+                return value;
+            };
+
+            switch (edge.useKind()) {
+            case Int32Use:
+                falseNodes.append({ edge.node(), constant(edge.node(), jsNumber(0)) });
+                break;
+            case DoubleRepUse: {
+                AbstractValue trueValue = forNode(edge.node());
+                // We know the input is not NaN if it has a truthy value.
+                if (trueValue.filter(~SpecDoubleNaN) == FiltrationOK)
+                    trueNodes.append({ edge.node(), trueValue });
```

Comment:
I think we can use `WTFMove ` for `trueNodes.append({ edge.node(), WTFMove(trueValue) });`

Reply:
Done.